### ISSUE TITLE
Cannot obtain symbol for type parameter of local function

### ISSUE THREAD

**Issue by JosefPihrt** [dotnet/roslyn#23699]
**Version Used**: VS 15.5 Community

**Steps to Reproduce**:
```csharp
class Foo
{
    void Bar()
    {
        void LocalFunction<T>(T parameter)
        {
        }
    }
}
```
I am unable to obtain symbol for `T` in type parameter list.

I can, however, obtain symbol for `T` in parameter.

**Comment by jcouv** [comment]
@JosefPihrt I took a quick look, but didn't see a problem. Can you clarify?

```C#
        [Fact]
        public void Bug()
        {
            var src = @"
class C
{
    void M()
    {
        void LocalFunction<T>(T parameter)
        {
        }
    }
}
";
            var comp = CreateStandardCompilation(src);
            var tree = comp.SyntaxTrees.Single();
            var model = comp.GetSemanticModel(tree);

            var localDecl = tree.FindNodeOrTokenByKind(SyntaxKind.LocalFunctionStatement);
            var localSymbol = Assert.IsType<LocalFunctionSymbol>(model.GetDeclaredSymbol(localDecl.AsNode()));

            Assert.Equal("T", localSymbol.TypeParameters[0].ToTestDisplayString());

            var parameterSymbol = localSymbol.Parameters[0];
            Assert.True(localSymbol.TypeParameters[0] == parameterSymbol.Type);
        }
```

**Comment by JosefPihrt** [comment]
`semanticModel.GetDeclaredSymbol(typeParameterOfLocalFunction)` returns null.

`semanticModel.GetDeclaredSymbol(typeParameterOfMethod)` returns symbol.

I assume it should return a symbol in both cases.

Also. when I use Visual Studio to Find All References of T (from `<T>` and not `T parameter`) it returns no results.

**Comment by jcouv** [comment]
Ah, I see. Let me check

**Comment by jcouv** [comment]
So, the semantic model seems to behave the same on type arguments of methods and local functions. See code below. So that part seems by-design.
That said, FAR does behave differently in the local function case than in the method case.

```C#
        [Fact]
        public void Bug()
        {
            var src = @"
class C
{
    void M<U>(U parameter1)
    {
        void LocalFunction<T>(T parameter2)
        {
        }
    }
}
";
            var comp = CreateStandardCompilation(src);
            var tree = comp.SyntaxTrees.Single();
            var model = comp.GetSemanticModel(tree);

            var methodDecl = (LocalFunctionStatementSyntax)tree.FindNodeOrTokenByKind(SyntaxKind.LocalFunctionStatement).AsNode();
            var tTypeParameter = methodDecl.TypeParameterList.Parameters[0];
            var tSymbol = model.GetDeclaredSymbol(tTypeParameter);
            Assert.Null(tSymbol);

            var tParameter = methodDecl.ParameterList.Parameters[0];
            var tSymbol2 = model.GetDeclaredSymbol(tParameter);
            Assert.Equal("T", tSymbol2.Type.ToTestDisplayString());

            var method = (MethodDeclarationSyntax)tree.FindNodeOrTokenByKind(SyntaxKind.MethodDeclaration).AsNode();
            var uTypeParameter = methodDecl.TypeParameterList.Parameters[0];
            var uSymbol = model.GetDeclaredSymbol(uTypeParameter);
            Assert.Null(uSymbol);
        }
```

**Comment by JosefPihrt** [comment]
There is a mistake in your code. Third line from the end should be

```csharp
var uTypeParameter = method.TypeParameterList.Parameters[0];
```
`uSymbol` is not null which means that the semantic model does NOT behave the same on type arguments of methods and local functions.

**Comment by jcouv** [comment]
Indeed. Thanks!

**Comment by jcouv** [comment]
Fix is pending review.

**Comment by JosefPihrt** [comment]
Thanks for a quick fix!

**Comment by jcouv** [comment]
Fix should become available in 15.6 preview 3.
